agent/agent.go

```diff
@@ -2340,13 +2340,13 @@ func (a *Agent) ReloadConfig(newCfg *Config) (bool, error) {
 	}
 
 	// Get the new client listener addr
-	httpAddr, err := newCfg.ClientListener(a.config.Addresses.HTTP, a.config.Ports.HTTP)
+	httpAddr, err := newCfg.ClientListener(newCfg.Addresses.HTTP, newCfg.Ports.HTTP)
 	if err != nil {
 		errs = multierror.Append(errs, fmt.Errorf("Failed to determine HTTP address: %v", err))
 	}
 
 	// Deregister the old watches
-	for _, wp := range a.config.WatchPlans {
+	for _, wp := range prevConfig.WatchPlans {
```

Comment:
I think we might want to eventually keep these on the agent itself and call `a.unloadWatches()` here, so we don't need to pass the old config, but this is a more minimal change so probably better for Tuesday.